mailcatcher.gemspec

```diff
@@ -37,8 +37,9 @@ Gem::Specification.new do |s|
   s.add_dependency "rack", "~> 1.5"
   s.add_dependency "sinatra", "~> 1.2"
   s.add_dependency "sqlite3", "~> 1.3"
-  s.add_dependency "thin", "~> 1.5.0"
-  s.add_dependency "skinny", "~> 0.2.3"
+  s.add_dependency "thin", "~> 1.8.0"
+  s.add_dependency "net-smtp", "~> 0.3.0"
```

Comment:
Why was net-smtp added here?

Reply:
> net-smtp

`net-smtp` has been extracted from the default gems in Ruby 3.1

Ref: 
- rails/rails#44083
- https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/

Reply:
As already mentioned by @tagliala, as of Ruby 3.1, net-smtp was moved to bundled gems.

Reply:
Ah, I see. Thanks.

Reply:
This spits a lot of warnings during startup for me on Ruby 2.7.5:

```
/opt/rbenv/versions/2.7.5/lib/ruby/2.7.0/net/protocol.rb:66: warning: already initialized constant Net::ProtocRetryError
/opt/rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/net-protocol-0.1.2/lib/net/protocol.rb:68: warning: previous definition of ProtocRetryError was here
/opt/rbenv/versions/2.7.5/lib/ruby/2.7.0/net/protocol.rb:206: warning: already initialized constant Net::BufferedIO::BUFSIZE
/opt/rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/net-protocol-0.1.2/lib/net/protocol.rb:208: warning: previous definition of BUFSIZE was here
/opt/rbenv/versions/2.7.5/lib/ruby/2.7.0/net/protocol.rb:503: warning: already initialized constant Net::NetPrivate::Socket
/opt/rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/net-protocol-0.1.2/lib/net/protocol.rb:504: warning: previous definition of Socket was here
```

Reply:
Tried ruby 2.7.5 and I'm getting same warnings. Not sure how to avoid that yet, will keep looking for a solution.

Reply:
Following this issue: https://github.com/ruby/net-imap/issues/16 , adding net-http as dependency solves "already initialized constant" issues with net-protocol.